### ISSUE TITLE
cssesc always escapes identifier characters above U+00A0

### ISSUE THREAD

**Issue by Kovensky** [mathiasbynens/cssesc#8]
As is not demonstrated by the online demo at https://mothereff.in/css-escapes.

If a `@charset "utf-8";` directive is present in the input, or an appropriate option is set in the options object, it would be nice to be able to preserve characters that don't need to be escaped.

The [CSS 2.1 spec](https://www.w3.org/TR/CSS21/syndata.html#characters) allows identifiers to contain characters above U+00A0, and at least CSS 2.0 allowed characters "161 and higher".

**Comment by mathiasbynens** [comment]
To only escape what’s absolutely necessary for modern browsers, you could use [`CSS.escape()`](https://mths.be/cssescape).

But I agree, cssesc should have an option to opt-in to similar behavior. Care to submit a pull request?

**Comment by mathiasbynens** [comment]
This suggestion goes against the goal of this library:

> A JavaScript library for escaping CSS strings and identifiers while generating the shortest possible ASCII-only output.

To only escape what’s absolutely necessary for modern browsers, you could use [`CSS.escape()`](https://mths.be/cssescape).

That said, cssesc _could_ have an option to opt-in to similar behavior. If someone submits a patch, I’ll consider it.

**Comment by Kovensky** [comment]
Indeed, including high characters should be an off-by-default option if provided at all. As for `css.escape`, there is unfortunately no way to use it without it writing to `global` :(

**Comment by chriseppstein** [comment]
I would like to have this option and will make a PR for it. The polyfil does not optimize string escapes, and I feel this library is a better place to have both capabilities so that a single dependency can accommodate libraries that need to handle a range of browser support preferences from users.

Do you have any preference for the option name (and corresponding CLI switch)? I suggest one of the following (in order of my most preferred to least preferred):

* `preserveUnicode` (defaults to `false`) 
* `onlySpecialChars` (defaults to `false`)
* `unicodeOutput` (defaults to `false`)
* `asciiOutput` (defaults to `true`)

**Comment by mathiasbynens** [comment]
I'd prefer the name `minimal`, to match jsesc: https://github.com/mathiasbynens/jsesc#minimal
